### ISSUE TITLE
navigation event listener missing declarations for `transitionStart` and `transitionEnd` event names

### ISSUE THREAD

**Issue by maltoze** [expo/expo#27557]
### Minimal reproducible example

navigation.addListener('transitionEnd', () => {});

### Which package manager are you using? (Yarn is recommended)

bun

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

navigation event listener missing declarations for `transitionStart` and `transitionEnd` event names.
https://reactnavigation.org/docs/native-stack-navigator#events

```js
  const [isReady, setIsReady] = useState(false);
  const navigation = useNavigation();

  useEffect(() => {
    // Argument of type '"transitionEnd"' is not assignable to parameter of type 'keyof EventMapCore<Readonly<{ key: string; index: number; routeNames: never[]; history?: unknown[]; routes: NavigationRoute<RootParamList, never>[]; type: string; stale: false; }>>'.
    navigation.addListener('transitionEnd', () => {
      setIsReady(true);
    });
  }, []);
```

### Environment

    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 18.19.0 - /usr/local/opt/nvm/versions/node/v18.19.0/bin/node
      Yarn: 1.22.21 - /usr/local/opt/nvm/versions/node/v18.19.0/bin/yarn
      npm: 10.2.3 - /usr/local/opt/nvm/versions/node/v18.19.0/bin/npm
      Watchman: 2023.12.04.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
      Android SDK:
        API Levels: 23, 28, 29, 30, 31, 33
        Build Tools: 28.0.3, 29.0.1, 29.0.2, 30.0.3
        System Images: android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-32 | Google Play Intel x86_64 Atom
    IDEs:
      Android Studio: 2023.1 AI-231.9392.1.2311.11076708
      Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.11 => 50.0.11 
      expo-router: ~3.4.8 => 3.4.8 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
      react-native-web: ~0.19.6 => 0.19.10 
    npmGlobalPackages:
      eas-cli: 7.3.0
    Expo Workflow: managed

**Comment by wcastand** [comment]
it's pretty sad will stop even making issues when they are valid and still get auto closed....

**Comment by dakkor71** [comment]
+ 1

**Comment by isaachinman** [comment]
This issue is still valid and should not be closed. It is not completed.

A reproduction _was_ provided.

cc @brentvatne

**Comment by hirbod** [comment]
The issue is not valid, as a reproduction was _not_ provided. A snack or a GitHub repo would've been helpful to quickly analyze this, but a simple code snippet is not enough.

Anyway, the type issue comes from @react-navigation/core and is not caused by Expo or Expo Router. You can see the relevant code [here](https://github.com/react-navigation/react-navigation/blob/f811b989035f36e9dbd6b56524dd9dfdaf93540d/packages/core/src/types.tsx#L117).

So, you'd need to file an issue upstream. Furthermore, `transitionStart` and `transitionEnd` have been buggy in the current Expo Router version, but this was recently fixed in react-navigation/native-stack 6.10.0. However, Expo Router is pinned to 6.9*, so you'd need to add a resolution as well.
